labellab-client/src/actions/image.js

```diff
@@ -40,21 +40,25 @@ export const submitImage = (data, projectId, callback) => {
   }
 }
 
-export const fetchProjectImage = (imageId, callback) => {
+export const fetchProjectImage = (projectId, imageId, callback) => {
   return dispatch => {
-    dispatch(request())
-    FetchApi.get('/api/v1/image/get_image/' + imageId)
-      .then(res => {
-        dispatch(success(res.data.body))
-        callback()
-      })
-      .catch(err => {
-        if (err.response) {
-          err.response.data
-            ? dispatch(failure(err.response.data.msg))
-            : dispatch(failure(err.response.statusText, null))
-        }
-      })
+    if (isNaN(imageId) || !Number.isInteger(parseFloat(imageId))) {
+      dispatch(failure('ImageId is not a number'))
+    } else {
+      dispatch(request())
```

Comment:
Move request dispatch outside the if-else block.
`dispatch(request())` will always trigger. 
Then depending on the rest of the request, either `dispatch(success())` or `dispatch(failure())`

Reply:
Done ✅